source/developers-guide/models/index.md

```diff
@@ -104,8 +103,8 @@ class ModelSubscriber implements EventSubscriber
     public function getSubscribedEvents()
     {
         return [
-           Events::prePersist,
-           Events::postPersist,
+           'Article::prePersist',
```

Comment:
I think both are valid. see https://github.com/shopware/shopware/blob/5.5/engine/Shopware/Models/Media/MediaSubscriber.php#L60

and then we should get the `LifecycleEventArgs`

Reply:
Yes, that's totally fine. You get the LifecycleEventArgs when defining the handlers direct in the model class. In that case you are using the the original doctrine event system. The MediaSubscriber implements the Doctrine\Common\EventSubscriber interface. This is the same as defining them in directly in the model class and, again, belongs to the doctrine event system. (https://github.com/shopware/shopware/blob/4fb20a661576322704ea0d20332bdbe3c49e2ab6/engine/Shopware/Components/DependencyInjection/services.xml#L308)

But in this example you are defining the event handlers in a shopware subscriber class. In this case you using shopware's event system. Doing so, you will get the shopware event class and not the doctrine event class.

Registering Events::prePersist to the event model doesn't fit to rest of the example. Within the handler the example refers to the article model. Thats the reason I changed Events to Article.

Reply:
Sorry. guess you are right. I didn't see the line '<tag name="doctrine.event_subscriber"/>' in the example. In my case I was using '<tag name="shopware.event_subscriber" />'. Which works to, but you will get the Enlight_Event_EventArgs instead of LifecycleEventArgs.